actions/go-flaky-tests/cmd/go-flaky-tests/github.go

```diff
@@ -0,0 +1,282 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os/exec"
+	"strings"
+	"text/template"
+	"time"
+)
+
+type DefaultGitHubClient struct {
+	config Config
+}
+
+func NewDefaultGitHubClient(config Config) *DefaultGitHubClient {
+	return &DefaultGitHubClient{config: config}
+}
+
+func (gh *DefaultGitHubClient) GetUsernameForCommit(commitHash string) (string, error) {
+	cmd := exec.Command("gh", "search", "commits", "--hash", commitHash, "--json", "author")
+
+	result, err := cmd.Output()
+	if err != nil {
+		return "", fmt.Errorf("failed to search commit: %w (output: %s)", err, string(result))
+	}
+
+	var response GitHubCommitSearchResponse
+
+	if err := json.Unmarshal(result, &response); err != nil {
+		return "", fmt.Errorf("failed to parse GitHub API response: %w", err)
+	}
+
+	if len(response) > 0 && response[0].Author.Login != "" {
+		return response[0].Author.Login, nil
+	}
+
+	return "", fmt.Errorf("no GitHub username found for commit %s; GitHub response %s", commitHash, string(result))
+}
+
+func (gh *DefaultGitHubClient) CreateOrUpdateIssue(test FlakyTest) error {
+	issueTitle := fmt.Sprintf("Flaky %s", test.TestName)
+
+	existingIssueURL, err := gh.SearchForExistingIssue(issueTitle)
+	if err != nil {
+		log.Printf("Warning: failed to search for existing issue: %v", err)
+	}
+
+	if existingIssueURL != "" {
+		log.Printf("📝 Found existing issue for %s, adding comment: %s", test.TestName, existingIssueURL)
+		return gh.AddCommentToIssue(existingIssueURL, test)
+	}
+
+	log.Printf("📝 Creating new issue for flaky test: %s", test.TestName)
+	issueBody, err := generateInitialIssueBody(test)
+	if err != nil {
+		return fmt.Errorf("failed to generate issue body: %w", err)
+	}
+
+	cmd := exec.Command("gh", "issue", "create",
+		"--repo", gh.config.Repository,
+		"--title", issueTitle,
+		"--body", issueBody,
+		"--label", "flaky-test")
+
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to create GitHub issue: %w, output: %s", err, string(output))
+	}
+
+	log.Printf("📝 Created GitHub issue: %s", strings.TrimSpace(string(output)))
+
+	issueURL := strings.TrimSpace(string(output))
+	return gh.AddCommentToIssue(issueURL, test)
+}
+
+func (gh *DefaultGitHubClient) SearchForExistingIssue(issueTitle string) (string, error) {
+	cmd := exec.Command("gh", "issue", "list",
```

Comment:
This will only work on the first 30 issues found if I'm not mistaken. Is that a problem for this use-case?